spec/presenters/two_factor_authentication/selection_presenter_spec.rb

```diff
@@ -25,6 +25,28 @@ def method
     it { expect(presenter.disabled?).to eq(false) }
   end
 
+  describe '#single_configuration_only?' do
+    it { expect(presenter.single_configuration_only?).to eq(false) }
+  end
+
+  describe '#mfa_added_label' do
```

Comment:
Not sure if this is still in-progress, but it'd be good to have `disabled?` tests here too:

```suggestion
  describe '#disabled?' do
    # ...
  end

  describe '#mfa_added_label' do
```

Reply:
That should be there up on line 24, right?
`describe '#disabled?' do`
`  it { expect(presenter.disabled?).to eq(false) }`
`end`

Reply:
Oh, you're right it's there. And it's passing for the default state. But the logic has changed quite a bit and we should have test coverage for each of the branching paths.

More specifically:

```
  describe '#disabled?' do
    let(:single_configuration_only) {}
    let(:mfa_configuration_count) {}

    before do
      allow(presenter).to receive(:single_configuration_only?).and_return(single_configuration_only)
      allow(presenter).to receive(:mfa_configuration_count).and_return(mfa_configuration_count)
    end

    context 'without single configuration restriction' do
      let(:single_configuration_only) { false }

      it 'is ...' do
        # ...
      end
    end

    context 'with single configuration only' do
      let(:single_configuration_only) { false }

      context 'with unimplemented mfa count (nil)' do
        let(:mfa_configuration_count) { nil }

        it 'is ...' do
          # ...
        end
      end

      context 'with no configured mfas' do
        let(:mfa_configuration_count) { 0 }

        it 'is ...' do
          # ...
        end
      end

      context 'with at least one configured mfa' do
        let(:mfa_configuration_count) { 1 }

        it 'is ...' do
          # ...
        end
      end
    end
  end

Reply:
I have added those in. I'm not sure how well executed. The last two contexts are coming up nil. Do I need more variables added to those cases?

Reply:
There was a typo in my suggestion:

```rb
    context 'with single configuration only' do
      let(:single_configuration_only) { false }
```

Should be `true`, to set the context that that method would return `true`.

I think there might have also been an issue in how the code was copied into the spec, since with the current code in the branch, the "with no configured mfas" and "with at least one configured mfa" test cases are not nested within the "with single configuration only" context, as they were in the suggestion above.

Reply:
Yes. I have fixed the nested structure to how you have suggested it. It seems to be working now.